core/src/Http/HttpRequest.cs

```diff
@@ -236,5 +238,32 @@ private static async Task<HttpResponse> CreateResponseAsync(HttpResponseMessage
                 StatusCode = response.StatusCode
             };
         }
+
+
+        private static async Task<HttpContent> CloneHttpContentAsync(HttpContent httpContent)
+        {
+            var temp = new MemoryStream();
+            await httpContent.CopyToAsync(temp).ConfigureAwait(false);
+            temp.Position = 0;
+            
+            var clone = new StreamContent(temp);
+            if (httpContent.Headers != null)
+            {
+                foreach (var h in httpContent.Headers)
+                {
+                    clone.Headers.Add(h.Key, h.Value);
+                }
+            }
+
+            // WORKAROUND 
+            // On UWP there is a bug in the Http stack that causes an exception to be thrown when moving around a stream.
+            // https://stackoverflow.com/questions/31774058/postasync-throwing-irandomaccessstream-error-when-targeting-windows-10-uwp
+            // LoadIntoBufferAsync is necessary to buffer content for multiple reads - see https://stackoverflow.com/questions/26942514/multiple-calls-to-httpcontent-readasasync
+            // Documentation is sparse, but it looks like loading the buffer into memory avoids the bug, without 
+            // replacing the System.Net.HttpClient with Windows.Web.Http.HttpClient, which is not exactly a drop in replacement
+            await clone.LoadIntoBufferAsync().ConfigureAwait(false);
```

Comment:
This line is the actual fix

Reply:
nice find! should there be an #if UWP for this then?

---
In reply to: [220696902](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1273#discussion_r220696902) [](ancestors = 220696902)

Reply:
Yes, I guess so. I'll address this in a separate PR.